new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java

```diff
@@ -3,30 +3,268 @@
 
 package oracle.weblogic.kubernetes.assertions.impl;
 
-import java.util.concurrent.Callable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
 
-public class Kubernetes {
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.Configuration;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodList;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.openapi.models.V1ServiceList;
+import io.kubernetes.client.util.ClientBuilder;
+import oracle.weblogic.kubernetes.extensions.LoggedTest;
 
-  public static Callable<Boolean> podExists(String podName, String domainUID, String namespace) {
-    return () -> {
-      return true;
-    };
+public class Kubernetes implements LoggedTest {
+
+  private static final String OPERATOR_NAME = "weblogic-operator-";
+
+  private static ApiClient apiClient = null;
+  private static CoreV1Api coreV1Api = null;
+  private static CustomObjectsApi customObjectsApi = null;
+  private static final String RUNNING = "Running";
+  private static final String TERMINATING = "Terminating";
+
+  static {
+    try {
+      Configuration.setDefaultApiClient(ClientBuilder.defaultClient());
+      apiClient = Configuration.getDefaultApiClient();
+      coreV1Api = new CoreV1Api();
+      customObjectsApi = new CustomObjectsApi();
+    } catch (IOException ioex) {
+      throw new ExceptionInInitializerError(ioex);
+    }
   }
 
-  public static Callable<Boolean> podRunning(String podName, String domainUID, String namespace) {
-    return () -> {
-      return true;
-    };
+  /**
+   * Checks if a pod exists in a given namespace in any state.
+   * @param namespace in which to check for the pod existence
+   * @param domainUid the label the pod is decorated with
+   * @param podName name of the pod to check for
+   * @return true if pod exists otherwise false
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static boolean isPodExists(String namespace, String domainUid, String podName) throws ApiException {
+    logger.info("Checking if the pod exists in namespace");
+    String labelSelector = null;
+    if (domainUid != null) {
+      labelSelector = String.format("weblogic.domainUID in (%s)", domainUid);
+    }
+    V1Pod pod = getPod(namespace, labelSelector, podName);
+    return pod != null;
```

Comment:
this is too magical. what does this mean?
do your comparison and assign the value to a well-named variable, then return the var